common/app/model/dotcomrendering/DotcomRenderingDataModel.scala

```diff
@@ -452,7 +455,14 @@ object DotcomRenderingDataModel {
       twitterHandle = content.tags.contributors.headOption.flatMap(_.properties.twitterHandle),
     )
 
+    def hasAffiliateLinks(
+        blocks: Seq[APIBlock],
+    ): Boolean = {
+      blocks.exists(block => DotcomRenderingUtils.stringContainsAffiliateableLinks(block.bodyHtml))
+    }
```

Comment:
Perhaps we should move this check up to a lazy val on the [`Article class`](https://github.com/guardian/frontend/blob/main/common/app/model/content.scala#L594), similar to how it works with [`Gallery`](https://github.com/guardian/frontend/blob/main/common/app/model/content.scala#L869-L870)